examples/middleware/notifications_producer.py

```diff
@@ -0,0 +1,41 @@
+import logging
+import requests
+import asyncio
+
+from miteD.middleware.api import api
+from miteD.middleware.methods import get
+from miteD.middleware.types import json
+
+
+@api(
+    name='middleware-produces-notifications',
+    versions=['1.0'],
+    broker_urls=['nats://127.0.0.1:4222'],
+    notification_topics=['updates'],
+    port=8001,
+)
+class MiddlewareProducesNotifications:
+    @get(path='/')
+    @json
+    async def root(self, request):
+        print(dir(self))
+        msg = 'Update notification from {}'.format(self.__class__.__name__)
+        await self.notify.updates(msg)
+        return 'ROOT'
+
+
+def do_request():
+    r = requests.get('http://127.0.0.1:8001/1.0/')
+    logging.info('Request returned: {}'.format(r.json()))
+
+
+if __name__ == '__main__':
+
+    logging.basicConfig(level=logging.DEBUG)
+    loop = asyncio.get_event_loop()
+    test = MiddlewareProducesNotifications()
+    loop.run_in_executor(None, do_request)
```

Comment:
You can either write a blocking function and run in executor or write async function and run in event loop. Either will work.

As for using asyncio.ensure_future() -> I'd rather use asyncio.create_task() as this is the preferred way of spawning tasks as per python 3.7 spec.

ascyncio.gather() is meant for aggregating results of several tasks rather than spawning them.

Reply:
Ok. Cool. Just making sure I understood what we were doing. Why _are_ we doing it this way rather than the `async def` way?